src/parsing/effect_parsing.cpp

```diff
@@ -339,6 +340,9 @@ void ef_scope_random_owned(token_generator& gen, error_handler& err, effect_buil
 
 		context.compiled_effect[payload_size_offset] = uint16_t(context.compiled_effect.size() - payload_size_offset);
 		context.limit_position = old_limit_offset;
+
+		auto const new_scope_size = simplify_effect(context.compiled_effect.data() + scope_offset);
```

Comment:
why are you running simplify effect on its own here? doesn't it get run on all effects as a last step anyways? if so, this should be unnecessary if it is working correctly

Reply:
optimizes random_owned so the overall size is smaller, because scopes store sizes in uint16_t, and at the time we finish the "main" scope, we already surpassed that limit - this is a workaround to make random_owned smaller

Reply:
right, but as your tests revealed, simplify *breaks* if you run it on a sequence of effects that are not in a scope, and that isn't that exactly what you are doing here?